src/lib/Keys.ts

```diff
@@ -64,6 +64,24 @@ export function readBase64WithPEM(content: string): Uint8Array {
   return decodeBase64(base64);
 }
 
+export const validateSignature = (
+  msg: Uint8Array,
+  signature: Uint8Array,
+  pk: CLPublicKey
+): boolean => {
+  if (pk.isEd25519()) {
+    return nacl.sign_detached_verify(msg, signature, pk.value());
+  }
+  if (pk.isSecp256K1()) {
+    return secp256k1.ecdsaVerify(
+      signature,
+      sha256(Buffer.from(msg)),
+      pk.value()
+    );
+  }
+  throw Error('Unsupported PublicKey type');
```

Comment:
Same here ? Not sure if it should throw an error or return false instead ?

Reply:
In practice, this error should never happen, because the `CLPublicKey` is correct when created. 
But in case something really untypical would happen (eg we will change the `CLPublicKey` api) it's better to get an error thrown IMHO.